dali/operators/image/color/equalize.cc

```diff
@@ -25,4 +30,77 @@ The supported inputs are images and videos of uint8_t type.)code")
     .InputLayout(0, {"HW", "HWC", "CHW", "FHW", "FHWC", "FCHW"})
     .AllowSequences();
 
+namespace equalize {
+
+class EqualizeCPU : public Equalize<CPUBackend> {
+ public:
+  explicit EqualizeCPU(const OpSpec &spec) : Equalize<CPUBackend>(spec) {}
+
+ protected:
+  void RunImpl(Workspace &ws) override {
+    const auto &input = ws.Input<CPUBackend>(0);
+    auto &output = ws.Output<CPUBackend>(0);
+    auto in_view = view<const uint8_t>(input);
+    auto out_view = view<uint8_t>(output);
+    int sample_dim = in_view.shape.sample_dim();
+    // by the check in Equalize::SetupImpl
+    assert(input.type() == type2id<uint8_t>::value);
+    // enforced by the layouts specified in operator schema
+    assert(sample_dim == 2 || sample_dim == 3);
+    output.SetLayout(input.GetLayout());
+    int num_samples = in_view.num_samples();
+    for (int sample_idx = 0; sample_idx < num_samples; sample_idx++) {
+      const auto &in_sample = in_view[sample_idx];
+      const auto &in_shape = in_sample.shape;
+      int64_t num_channels = sample_dim == 2 ? 1 : in_shape[2];
+      DALI_ENFORCE(
+          1 <= num_channels && num_channels <= CV_CN_MAX,
+          make_string("The CPU equalize operator supports images with number of channels in [1, ",
+                      CV_CN_MAX, "] channels. However, the sample at index ", sample_idx, " has ",
+                      num_channels, " channels."));
+      DALI_ENFORCE(in_shape[0] <= std::numeric_limits<int>::max() &&
+                       in_shape[1] <= std::numeric_limits<int>::max(),
+                   make_string("The image height and width must not exceed the ",
+                               std::numeric_limits<int>::max(), ". However, the sample at index ",
+                               sample_idx, " has shape ", in_shape, "."));
+    }
+    auto &tp = ws.GetThreadPool();
+    for (int sample_idx = 0; sample_idx < num_samples; sample_idx++) {
+      tp.AddWork([this, sample_idx, &in_view, &out_view](int) {
+        RunSample(in_view, out_view, sample_idx);
+      });
+    }
+    tp.RunAll();
+  }
+
+  template <int ndim>
+  void RunSample(TensorListView<StorageCPU, const uint8_t, ndim> &in_view,
+                 TensorListView<StorageCPU, uint8_t, ndim> &out_view, int sample_idx) {
```

Comment:
nitpick: Accept TensorView's instead of TLV, and get rid of sample_idx argument.

Reply:
done